### PR TITLE
ENG-4839 Add advertiseOnDiscoverPage Flag to Provider Model and Admin Interface

### PR DESCRIPTION
**Pull request by uditijmehta**
## Purpose

Adding a boolean attribute to the AbstractProvider model. This change enables administrators to control whether a provider is displayed on the OSF Discover pages directly from the admin interface.

## Changes

- Added advertiseOnDiscoverPage boolean field to AbstractProvider
- Updated the admin forms and templates to include a checkbox for advertiseOnDiscoverPage
- Adjusted the PreprintProviderDisplay view to serialize the new advertiseOnDiscoverPage field

## Ticket

<https://openscience.atlassian.net/browse/ENG-4839>

### CODE REVIEW

**Comment by Johnetordoff** [comment]
Seems like you should also include a migration file, one created by running `python3 manage.py makemigrations`.

File: osf/models/provider.py [hunk 1]
```diff
@@ -122,6 +122,7 @@ def update_or_create_from_json(cls, provider_data, user):
     primary_collection = models.ForeignKey('Collection', related_name='+',
                                            null=True, blank=True, on_delete=models.SET_NULL)
     name = models.CharField(null=False, max_length=128)  # max length on prod: 22
+    advertiseOnDiscoverPage = models.BooleanField(default=True, help_text="Indicates if the provider should be advertised on the discovery page.")
```
Comment (Johnetordoff) [inline osf/models/provider.py:125]: Use underscores and no caps for `advertiseOnDiscoverPage` here please, it's ok in the HTML, but not on the backend. It should follow the convention of the other attribute names.

File: api/providers/serializers.py [hunk 2]
```diff
@@ -203,6 +203,7 @@ class Meta:
         'share_publish_type',
         'reviews_workflow',
         'permissions',
+        'advertise_on_discover_page',
```
Comment (brianjgeiger) [inline api/providers/serializers.py:206]: You've made this filterable, which is good, but you still have to add the field to the serializer to get it serialized in the API response. Somewhere around L213 below.

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
Almost there. You still need to serialize the field in the API.

**Comment by brianjgeiger** [comment]
Also it looks like you have too many blank lines, according to Flake8:

```
Trim Trailing Whitespace.................................................Failed
hookid: trailing-whitespace

Files were modified by this hook. Additional output:

Fixing admin/preprint_providers/views.py

Flake8...................................................................Failed
hookid: flake8

admin/preprint_providers/forms.py:33:5: E303 too many blank lines (2)
```

File: osf/migrations/0020_auto_20240508_1455.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+            model_name='abstractprovider',
+            name='advertise_on_discover_page',
+            field=models.BooleanField(default=True),
+        ),
+        migrations.AlterField(
+            model_name='abstractprovider',
+            name='advisory_board',
+            field=models.TextField(blank=True, default='', help_text='Indicates if the provider should be advertised on the discovery page.'),
+        ),
+    ]
```
Comment (brianjgeiger) [inline osf/migrations/0020_auto_20240508_1455.py:23]: I think you need to delete this migration.

**Review by brianjgeiger — APPROVED** [review]

**Pull request merged**